invenio_rdm_records/services/pids/providers/datacite.py

```diff
@@ -87,7 +88,8 @@ def reserve(self, record, pid):
         # Only registered PIDs can be updated.
         try:
             pid.reserve()
-            self.client.draft_doi(metadata=record, doi=pid.pid_value)
+            doc = RDMDataCite43JSONSerializer.serialize_object(record)
+            self.client.draft_doi(metadata=doc, doi=pid.pid_value)
```

Comment:
Question: How are we dealing with restricted records and DOIs? how are we dealing with fully embargoed records?

Thoughts:
- The base DataCite provider should not register the draft DOI with DataCite on the reserve, but only on publish.
- Fully embargoed records: regsiter the DOI upon the embargo being lifted.

Reply:
> How are we dealing with restricted records and DOIs? how are we dealing with fully embargoed records? 

Opened an issue to treat this: https://github.com/inveniosoftware/invenio-rdm-records/issues/541

> Fully embargoed records: regsiter the DOI upon the embargo being lifted.

Same issue as above

> The base DataCite provider should not register the draft DOI with DataCite on the reserve, but only on publish.

I agree on this, but `draft_doi` is not registering... just reserving the DOI. It can be **hard** deleted from datacite. Is it possible to *need* to reserve the DOI on DataCite? e.g. several instances using the same provider/credentials might want to avoid clashes? Not sure if this can happen...